tests/unit/test_config.py

```diff
@@ -109,3 +121,136 @@ def test_profile_config_option(
+    return_value="10.10.10.10:10009",
+)
+@patch(
+    "managers.service.ServiceManager.reconcile_services",
+    return_value=True,
+)
+@patch("config.spark.SparkConfig._get_spark_master", return_value="k8s://https://spark.master")
+@patch("managers.k8s.K8sManager.is_s3_configured", return_value=True)
+@patch(
+    "config.spark.SparkConfig._sa_conf", return_value={"spark.hadoop.fs.s3a.endpoint": "foo.bar"}
+)
+@pytest.mark.parametrize(
+    "k8s_selectors",
+    [
+        ("a:b", [("spark.kubernetes.node.selector.a", "b")]),
+        (
+            "a:b,c:d",
+            [("spark.kubernetes.node.selector.a", "b"), ("spark.kubernetes.node.selector.c", "d")],
+        ),
+    ],
+)
+def test_k8s_node_selectors_config_option(
+    mock_sa_conf,
+    mock_s3_configured,
+    mock_get_master,
+    mock_reconcile_service,
+    mock_service_endpoint,
+    mock_update_client_endpoints,
+    mock_valid_sa,
+    mock_valid_ns,
+    mock_user_exists,
+    mock_set_password,
+    kyuubi_context: Context,
+    kyuubi_container: Container,
+    spark_service_account_relation: Relation,
+    auth_db_relation: Relation,
+    kyuubi_peers_relation: PeerRelation,
+    tmp_path,
+    k8s_selectors,
+) -> None:
+    """Test profile config option."""
+    logger.info(f"{k8s_selectors[0]}")
+    logger.info(f"{k8s_selectors[1]}")
+    state = State(
+        relations=[spark_service_account_relation, auth_db_relation, kyuubi_peers_relation],
+        containers=[kyuubi_container],
+        config={"k8s-node-selectors": k8s_selectors[0]},
+        leader=True,
+    )
+    out = kyuubi_context.run(kyuubi_context.on.config_changed(), state)
+    assert out.unit_status == Status.ACTIVE.value
+
+    spark_properties = parse_spark_properties(tmp_path)
+    logger.info(f"profile: testing => spark_properties: {spark_properties}")
+
+    for k, v in k8s_selectors[1]:
+        assert k in spark_properties
+        assert spark_properties[k] == v
+
+
+@patch("managers.auth.AuthenticationManager.set_password")
+@patch("managers.auth.AuthenticationManager.user_exists", return_value=True)
+@patch("managers.k8s.K8sManager.is_namespace_valid", return_value=True)
+@patch("managers.k8s.K8sManager.is_service_account_valid", return_value=True)
+@patch(
+    "events.provider.KyuubiClientProviderEvents.update_clients_endpoints",
+    return_value=True,
+)
+@patch(
+    "managers.service.ServiceManager.get_service_endpoint",
+    return_value="10.10.10.10:10009",
+)
+@patch(
+    "managers.service.ServiceManager.reconcile_services",
+    return_value=True,
+)
+@patch("config.spark.SparkConfig._get_spark_master", return_value="k8s://https://spark.master")
+@patch("managers.k8s.K8sManager.is_s3_configured", return_value=True)
+@patch(
+    "config.spark.SparkConfig._sa_conf", return_value={"spark.hadoop.fs.s3a.endpoint": "foo.bar"}
+)
+@pytest.mark.parametrize(
+    "k8s_selectors",
+    [
+        ("ab", [("spark.kubernetes.node.selector.a", "b")]),
```

Comment:
Can we add more examples of invalid configs, specifically involving usage of non alphanumeric char in the key, and the value containing commas?

some ideas:
```
foo-bar : my-val
foo-bar : my,val
foo:bar:grok
foo:bar,:grok
```